payroll/models/hr_payslip.py

```diff
@@ -532,25 +532,24 @@ def _compute_payslip_line(self, rule, localdict, lines_dict):
         self.ensure_one()
         # check if there is already a rule computed with that code
         previous_amount = rule.code in localdict and localdict[rule.code] or 0.0
-        # compute the amount of the rule
-        amount, qty, rate, computed_name = rule._compute_rule(localdict)
-        rule_total = amount * qty * rate / 100.0
+        # compute the values of the rule
+        values = rule._compute_rule(localdict)
+        total = values["quantity"] * values["rate"] * values["amount"] / 100.0
+        values["total"] = total
         # set/overwrite the amount computed for this rule in the localdict
-        localdict[rule.code] = rule_total
+        localdict[rule.code] = total
         localdict["rules"].dict[rule.code] = rule
-        localdict["result_rules"].dict[rule.code] = BaseBrowsableObject(
-            {"quantity": qty, "rate": rate, "amount": amount, "total": rule_total}
-        )
+        localdict["result_rules"].dict[rule.code] = BaseBrowsableObject(values)
```

Comment:
I think you are leaving out including "total" in "result_rules". Original code is:

`        localdict["result_rules"].dict[rule.code] = BaseBrowsableObject(
            {"quantity": qty, "rate": rate, "amount": amount, "total": rule_total}
        )` 

It's okay for me to make a dict, but you should add the value values["total"] to values dict.